ouroboros-network/src/Ouroboros/Network/Diffusion/P2P.hs

```diff
@@ -794,6 +811,8 @@ runM Interfaces
         }
 
       publicStateVar <- newTVarIO emptyPublicPeerSelectionState
+      debugStateVarA <- newTVarIO $ emptyPeerSelectionState fuzzRng []
+      debugStateVarB <- newTVarIO $ emptyPeerSelectionState fuzzRng []
```

Comment:
I think we can have just one `TVar` we never run both `initiator-only` and `initiator-and-responder` modes at the same time.

Reply:
They are two different variables because they are different types.

Reply:
I've moved the creation of the variables so it is only one variable is created depending on mode.